samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -110,17 +108,73 @@ public ZkKeyBuilder getKeyBuilder() {
    * @return String representing the absolute ephemeralPath of this client in the current session
    */
   public synchronized String registerProcessorAndGetId(final ProcessorData data) {
+    String processorId = data.getProcessorId();
     if (ephemeralPath == null) {
-      ephemeralPath =
-          zkClient.createEphemeralSequential(
-              keyBuilder.getProcessorsPath() + "/", data.toString());
-
-      LOG.info("newly generated path for " + data +  " is " +  ephemeralPath);
-      return ephemeralPath;
+      /**
```

Comment:
No value addition in documenting, since SamzaException is unchecked and stream processor should die(Not a recoverable exception).

New checked exception isn't added for consistency( majority of codebase uses this for custom exception).